### PR TITLE
JDK-8302360  Atomic*.compareAndExchange Javadoc unclear

### PR DESCRIPTION
**Pull request by viktorklang-ora**
I think the following proposal is very non-invasive and merely draws attention to the fact that "witness value" is a specific term related to the notion of these atomic methods.

It's a small change which I think provides additional clarity, see JBS for the discussion on the topic.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302360](https://bugs.openjdk.org/browse/JDK-8302360): Atomic*.compareAndExchange Javadoc unclear

### Reviewers
 * [Martin Buchholz](https://openjdk.org/census#martin) (@Martin-Buchholz - **Reviewer**) ⚠️ Review applies to [923d23d0](https://git.openjdk.org/jdk/pull/12880/files/923d23d0cf3229aaeeebb57827a3930b36ff1ed9)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12880/head:pull/12880` \
`$ git checkout pull/12880`

Update a local copy of the PR: \
`$ git checkout pull/12880` \
`$ git pull https://git.openjdk.org/jdk pull/12880/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12880`

View PR using the GUI difftool: \
`$ git pr show -t 12880`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12880.diff">https://git.openjdk.org/jdk/pull/12880.diff</a>

</details>

### CODE REVIEW

**Review by Martin-Buchholz — APPROVED** [review]

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java [hunk 1]
```diff
@@ -353,8 +353,8 @@ public final void setRelease(V newValue) {
      *
      * @param expectedValue the expected value
      * @param newValue the new value
-     * @return the witness value, which will be the same as the
-     * expected value if successful
+     * @return the <em>witness value</em>, which is the current value
+     * at the time of the operation.
```
Comment (dholmes-ora) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java:357]: This change is not appropriate as discussed in the PR. And unclear why you changed only this version? Leftover?

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java (hunk 1 above)
Comment (viktorklang-ora) [reply src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java:357]: @dholmes-ora Good catch, will fix.

**Review by dholmes-ora — APPROVED** [review]
Looks fine now. Thanks.

**Comment by viktorklang-ora** [comment]
@AlanBateman Who would be willing to sponsor this? 🤔

**Pull request closed without merging**